infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -1285,39 +1174,24 @@ private void deleteFilesMatchingFilter(FileFilter fileFilter) {
     @Override
     public void reset() {
         // Trim all segments
-        long endSegment = Math.max(logMetadata.getGlobalTail(), 0L) / RECORDS_PER_LOG_FILE;
+        long endSegment = getEndSegment();
         log.warn("Global Tail:{}, endSegment={}", logMetadata.getGlobalTail(), endSegment);
 
         // Close segments before deleting their corresponding log files
-        closeSegmentHandlers(endSegment);
+        segmentSupervisor.closeSegmentHandlers(endSegment);
 
-        deleteFilesMatchingFilter(file -> {
-            try {
-                String segmentStr = file.getName().split("\\.")[0];
-                return Long.parseLong(segmentStr) <= endSegment;
-            } catch (Exception e) {
-                log.warn("reset: ignoring file {}", file.getName());
-                return false;
-            }
-        });
+        deleteFiles(endSegment, "reset: ignoring file {}");
```

Comment:
Can't you just push this to `SegmentSupervisor` and have a `deleteAllSegments?`

Reply:
Unfortunately, I can't because It uses logSizeQuota. I wouldn't like to add logsizeQuota as a dependency into SegmentSupervisor
At least not right now